mapillary/controller/save.py

```diff
@@ -13,38 +13,125 @@
+            # Enforce the header for field_names
+            field_names = (
+                ["ID", "WKT"]
+                + [
+                    key
+                    for key in features[0].keys()
+                    if key != "id" and key != "geometry"
+                ]
+                # organization_id may or may not exist in the flattened features
+                # If it does exist, then its value will be reflected
+                # If it does not exist, then it will be set to "NULL"
+                + ["organization_id"]
+            )
+
+            # Create the csv writer
+            writer = csv.DictWriter(file_path, fieldnames=field_names)
+
+            # Write the header
+            writer.writeheader()
+
+            for feature in features:
+                writer.writerow(
+                    {
+                        # ID is the id of the feature. It will always exist 
+                        # and will always be the first column
+                        "ID": feature["id"],
+
+                        # WKT is the geometry of the feature in well-known text format.
+                        # It will always exist
+                        "WKT": shape(feature["geometry"]).wkt,
+
+                        # The rest of the columns are the features attributes
+                        **{
+                            key: feature[key]
+                            for key in feature.keys()
+                            if key != "id" and key != "geometry"
+                        },
+
+                        # organization_id is the id of the organization that the feature
+                        # If it does exist, then its value will be reflected
+                        # If it does not exist, then it will be set to "NULL"
+                        "organization_id": feature["organization_id"]
+                        if "organization_id" in feature
+                        else "NULL",
+                    }
+                )
+    except Exception as e:
+        # If there is an error, log it
+        print(f"An error occured: {e}")
+    return None
+
+
+def save_as_geojson_controller(data: str, path: str, file_name: str) -> None:
     """Save data as GeoJSON to given file path
 
     :param data: The data to save as GeoJSON
-    :type data: dict
+    :type data: str
 
     :param path: The path to save to
     :type path: str
 
-    :return: Confirmation of whether the object was saved properly
-    :rtype: bool
-    """
-
-    # TODO: Requirement# 12B
+    :param file_name: The file name to save as
+    :type file_name: str
 
-    return {"Message": "Hello, World!"}
+    :return: Npne
+    :rtype: None
+    """
+    # Ensure that the geojson is a dictionary
+    if isinstance(data, str):
+        data = json.loads(data)
+    
+    try:
+        # Set the file name according to the given value. Default is "geojson.geojson"
+        file_name = file_name if file_name is not None else "geojson.geojson"
```

Comment:
same here, at the least I think `mapillary.geojson` so they know it's from mapillary if losing track later (I do this a lot with geojson.io, they call it map.geojson and I forget where the data came fro) or maybe a timestamp to differentiate and prevent overwrite of the same one.